tasks/watch_notify_same_primary.py

```diff
@@ -67,7 +69,29 @@ def start_watch(n):
             stdout=StringIO(),
             stderr=StringIO(),
             wait=False)
-    watches = [start_watch(i) for i in range(20)]
+        return proc
+
+    num = 20
+
+    watches = [start_watch(i) for i in range(num)]
+
+    # wait for them all to register
+    for i in range(num):
+        with safe_while() as proceed:
+            while proceed():
+                proc = remote.run(
+                    args = [
+                        "rados",
+                        "-p", pool,
+                        "listwatchers",
+                        obj(i)],
+                    stdout=StringIO())
+                lines = proc.stdout.getvalue()
+                n = lines.count('watcher=')
+                log.info('i see %d watchers for %s', n, obj(i))
+                if n >= 1:
+                    break
+                time.sleep(1)
```

Comment:
the `safe_while` utility allows you to define how many seconds to sleep (and other options). By default it will try for 10 times sleeping 6 seconds in between.

Was this intended to add another second to `time.sleep` ?